rfd/XXXX-tsh-upgrade.md

```diff
@@ -0,0 +1,117 @@
+---
+authors: Bernard Kim (bernard@goteleport.com)
+state: draft
+---
+# RFD XXXX - tsh upgrade
+
+## Required Approvers
+
+* Engineering: ...
+* Infra: ...
+* Kube: ...
+
+## What
+
+A tsh subcommand that will prompt the user to upgrade if the version of tsh is older than the cluster version.
+
+## Why
+
+One of our goals for Teleport Cloud is to increase the cadence at which clusters can be upgraded.
+This requires that Teleport Agents and Teleport client tools be up to date with the latest compatible version of the cluster.
```

Comment:
Is the automatic tsh upgrade process design for Cloud only or do we want to support self-hosted users as well?

Reply:
I think it should be supported for self-hosted as well. Doesn't feel like there would be much difference as the client will be checking the version against the proxy endpoint.